app/services/foreman/provision/ssh.rb

```diff
@@ -77,32 +77,14 @@ def initiate_connection!
         Timeout.timeout(8) do
           ssh.run('pwd')
         end
-      rescue Errno::ECONNREFUSED
-        logger.debug "Connection refused for #{address}, retrying"
-        sleep(2)
-        retry
-      rescue Errno::EHOSTUNREACH
-        logger.debug "Host unreachable for #{address}, retrying"
-        sleep(2)
-        retry
-      rescue Net::SSH::Disconnect
-        logger.debug "Host dropping connections for #{address}, retrying"
-        sleep(2)
-        retry
-      rescue Net::SSH::ConnectionTimeout
-        logger.debug "Host timed out for #{address}, retrying"
-        sleep(2)
-        retry
-      rescue Net::SSH::AuthenticationFailed
-        logger.debug "Auth failed for #{username} at #{address}, retrying"
+      rescue => e
```

Comment:
This will hide other exceptions which may be caused by Foreman itself, the list of exceptions we had here was there in purpose so that other exceptions don't `retry` but fail and the backtrace is shown.